apps/confidential/confidential.go

```diff
@@ -363,12 +363,36 @@ func (cca Client) AuthCodeURL(ctx context.Context, clientID, redirectURI string,
 	if err != nil {
 		return "", err
 	}
+	ap.State = o.state
 	ap.Claims = o.claims
 	ap.LoginHint = o.loginHint
 	ap.DomainHint = o.domainHint
 	return cca.base.AuthCodeURL(ctx, clientID, redirectURI, scopes, ap)
 }
 
+// WithState adds a user-generated state to the request.
+func WithState(state string) interface {
```

Comment:
should this not be `any` or `interface{}`, allowing for any kind of object to be attached.

Reply:
No, the `ap.State` type was already a string type and this PR merely adds a way to set that value so any change to the underlying types is outside the scope of this change.
Additionally, `ap.State` needs to be added as a query param and therefore URL encoded.

As a workaround you can just stringify whatever type you wish to add and pass it that way.